v2/processor.go

```diff
@@ -118,6 +120,43 @@ func NewPanicHandler(handler HandlerFunc) HandlerFunc {
 	}
 }
 
+// NewTracingHandler extracts the context from the message Application property if available, or from the existing context if not, and starts a span
+func NewTracingHandler(handler HandlerFunc) HandlerFunc {
+	return func(ctx context.Context, settler MessageSettler, message *azservicebus.ReceivedMessage) {
+		var err error
+		if message != nil && message.ApplicationProperties != nil {
+			ctx, err = ExtractMessageContext(message)
+			if err != nil {
+				// do something
+			}
+		}
+
+		ctx, span := tab.StartSpan(ctx, "")
+		if message != nil {
+			span.AddAttributes(
+				tab.StringAttribute("message.id", message.MessageID),
+				tab.StringAttribute("message.correlationId", *message.CorrelationID))
+			if message.ScheduledEnqueueTime != nil {
+				span.AddAttributes(tab.StringAttribute("message.scheduledEnqueuedTime", message.ScheduledEnqueueTime.String()))
+			}
+			if message.TimeToLive != nil {
+				span.AddAttributes(tab.StringAttribute("message.ttl", message.TimeToLive.String()))
+			}
+		} else {
+			span.Logger().Error(fmt.Errorf("message is nil"))
```

Comment:
Maybe Warn, instead of Error.

Reply:
Okay, I didn't see a warn method for the span logger so I used info and just included warning in the message